scripts/test/Muon/muon_context_test.py

```diff
@@ -0,0 +1,122 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+import sys
+import unittest
+from Muon.GUI.Common.muon_context import MuonContext
+from Muon.GUI.Common.observer_pattern import Observer
+from Muon.GUI.Common.muon_group import MuonGroup
+from Muon.GUI.Common.muon_pair import MuonPair
+import sys
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+from Muon.GUI.Common.utilities.load_utils import load_workspace_from_filename
+from Muon.GUI.Common.muon_data_context import MuonDataContext
+from Muon.GUI.Common.muon_gui_context import MuonGuiContext
+from Muon.GUI.Common.muon_group_pair_context import MuonGroupPairContext
+from mantid.api import AnalysisDataService
+import unittest
+from Muon.GUI.Common.observer_pattern import Observer
+from mantid.api import FileFinder
+from mantid.dataobjects import Workspace2D
+
+if sys.version_info.major < 2:
+    from unittest import mock
+else:
+    import mock
+
+class MuonContextTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        pass
+
+    def setUp(self):
+        AnalysisDataService.clear()
+        self.filepath = FileFinder.findRuns('EMU00019489.nxs')[0]
+        self.load_result, self.run_number, self.filename = load_workspace_from_filename(self.filepath)
+        self.loaded_data = MuonLoadData()
+        self.data_context = MuonDataContext(self.loaded_data)
+        self.gui_context = MuonGuiContext()
+        self.group_pair_context = MuonGroupPairContext()
+        self.gui_context.update({'RebinType': 'None'})
+
+        self.context = MuonContext(muon_data_context=self.data_context, muon_gui_context=self.gui_context, muon_group_context=self.group_pair_context)
+
+        self.data_context.instrument = 'EMU'
+
+        self.loaded_data.add_data(workspace=self.load_result, run=[self.run_number], filename=self.filename,
+                                  instrument='EMU')
+        self.data_context.current_runs = [[self.run_number]]
+        self.data_context.update_current_data()
+        self.group_pair_context.reset_group_and_pairs_to_default(self.load_result['OutputWorkspace'][0]._workspace,
+                                                                 'EMU', '')
+
+    def test_reset_groups_and_pairs_to_default(self):
+        self.assertEquals(self.group_pair_context.group_names, ['fwd', 'bwd'])
+        self.assertEquals(self.group_pair_context.pair_names, ['long'])
+
+    def test_calculate_group_calculates_group_for_given_run(self):
+        counts_workspace, asymmetry_workspace = self.context.calculate_group('fwd', run=[19489])
+
+        self.assertEquals(type(counts_workspace), Workspace2D)
+        self.assertEquals(type(counts_workspace), Workspace2D)
```

Comment:
should we not check the values from the calculation?

Reply:
All I am really interested in here is that the calculation is performed and gives a return of the correct type. I think validating the calculation is done correctly is better done on the tests for the relevant algorithms.

Reply:
I was thinking more to check we passed the correct inputs to the calculation. Suppose that would be covered by the system tests we discussed.